agent/config/builder.go

```diff
@@ -1969,15 +1969,30 @@ func (b *builder) cidrsVal(name string, v []string) (nets []*net.IPNet) {
 	return
 }
 
-func (b *builder) tlsCipherSuites(name string, v *string) []uint16 {
+func (b *builder) tlsVersion(name string, v *string) types.TLSVersion {
+	// TODO: should the nil check be moved into ParseTLSVersion and the arguments
+	// switched to take a pointer?
+	if v == nil {
+		return types.TLSVersionAuto
+	}
+
+	a, err := tlsutil.ParseTLSVersion(*v)
+	if err != nil {
+		// TODO: should a warning for deprecated config values be surfaced here somehow?
```

Comment:
This should probably be done in `applyDeprecatedTLSConfig`:

https://github.com/hashicorp/consul/blob/6db469699dbb25cd400acc6415ad0da4352ec2ab/agent/config/deprecated.go#L178-L267